cognite/client/_api/files.py

```diff
@@ -612,19 +612,52 @@ def retrieve_download_urls(
         results = tasks_summary.joined_results(unwrap_fn=lambda res: res.json()["items"])
         return {result.get("id") or result["externalId"]: result["downloadUrl"] for result in results}
 
+    @staticmethod
+    def _create_unique_file_names(file_names_in: list[str] | list[Path]) -> list[str]:
```

Comment:
This produces weird results when there's multiple suffixes:
```
["archive.tar.gz", "archive.tar.gz"]
--> file.tar(1).gz
```

Reply:
I unresolved this one as it still gives me weird results. If you disagree, let me know 😉

Reply:
I do actually disagree here, as this will occur very infrequently and when it does the soluton still works ok. If we need to define file extensions differently than the pathlib suffix function, I think the complexity of the codebase becomes unnecessary complex and not yield much value. But if you have a clever solution, I would love to see it :)

Reply:
Maybe this can be used?
```
>>> Path("archive.tar.gz").suffixes
['.tar', '.gz']
```

Reply:
Good call! I'll be back...